src/backtrace.c

```diff
@@ -169,13 +174,24 @@ static bool is_valid_address(uint32_t addr)
  * 
  * If not found, return a null header.
  */
-static symtable_header_t symt_open(void) {
-    if (SYMT_ROM == 0xFFFFFFFF) {
-        SYMT_ROM = rompak_search_ext(".sym");
-        if (!SYMT_ROM)
-            debugf("backtrace: no symbol table found in the rompak\n");
+static symtable_header_t symt_open(void *addr) {
+    dl_module_t *module = __dl_get_module(addr);
```

Comment:
Is it possible to reverse the logic here? First have a check if the address falls within the main app text segment, and only call __dl_get_module if it doesn't?